Fix duplicate function declarations in model package

## Description

After merging the model field compatibility fixes, there are duplicate function declarations between `converters.go` and `utils.go`. This causes a build error:

```
internal/domain/models/utils.go:13:6: BytesToString redeclared in this block
        internal/domain/models/converters.go:155:6: other declaration of BytesToString
internal/domain/models/utils.go:21:6: StringToBytes redeclared in this block
        internal/domain/models/converters.go:150:6: other declaration of StringToBytes
```

## Specific Issues

The following functions are duplicated:
- `BytesToString` is defined in both files
- `StringToBytes` is defined in both files

## Acceptance Criteria

1. Remove duplicate function declarations
2. Consolidate utility functions into a single file
3. Update any references to use the correct functions
4. Ensure the application builds and runs correctly

## Priority
High - This is blocking the build

--- a/internal/domain/models/converters.go
+++ b/internal/domain/models/converters.go
@@ -146,16 +146,6 @@ func ParseHTTPHeaders(headers map[string]string) []HTTPHeader {
 	return result
 }
 
-// StringToBytes converts a string to a byte slice
-func StringToBytes(s string) []byte {
-	return []byte(s)
-}
-
-// BytesToString converts a byte slice to a string
-func BytesToString(b []byte) string {
-	return string(b)
-}
-
 // ConvertHeadersToMap converts a slice of HTTPHeader to a map
 func ConvertHeadersToMap(headers []HTTPHeader) map[string]string {
 	result := make(map[string]string)

